transport/src/main/java/io/netty/channel/ChannelConfig.java

```diff
@@ -253,4 +259,11 @@
      * to detect the size of a message.
      */
     ChannelConfig setMessageSizeEstimator(MessageSizeEstimator estimator);
+
+    /**
+     * Set the {@link WriteBufferWaterMark} which is used for setting the high and low
+     * water mark of the write buffer.
+     */
+    ChannelConfig setWriteBufferWaterMark(WriteBufferWaterMark writeBufferWaterMark);
```

Comment:
@normanmaurer  why, I see the SocketChannelConfig is the interface extends ChannelConfig and hadn't override the old methods such as setWriteBufferLowWaterMark(int). so why need to override setWriteBufferWaterMark(WriteBufferWaterMark)?